classes/sota_freedom-u540.bbclass

```diff
@@ -0,0 +1,16 @@
+# Support fitImage by default
+KERNEL_CLASSES_sota = " kernel-fitimage "
+KERNEL_IMAGETYPE_sota = "fitImage"
+KERNEL_IMAGETYPES_remove_sota = "uImage"
+OSTREE_KERNEL = "${KERNEL_IMAGETYPE}-${INITRAMFS_IMAGE}-${MACHINE}-${MACHINE}"
+
+OSTREE_BOOTLOADER ?= "u-boot"
+INITRAMFS_FSTYPES = "cpio.gz"
+PREFERRED_PROVIDER_virtual/bootloader_sota ?= "u-boot"
+
+IMAGE_BOOT_FILES_sota ?= "fw_payload.bin boot.scr uEnv.txt"
```

Comment:
Could we change this to:
```
IMAGE_BOOT_FILES_append = " uEnv.txt ${OSTREE_KERNEL}"
```
That way we don't change what is already there.

Reply:
The kernel is available as part of the rootfs and not from the boot partition, which is why we have boot.scr and uEnv.txt, so that would only work if not really using ostree.

Reply:
So in this case the only pieces required at the boot partition is fw_payload.bin boot.scr uEnv.txt (in really only fw_payload.bin, but you would need to change the default u-boot load script).

Reply:
Ok. can it still be an append?

Reply:
Append will cause the kernel images to be available in the boot partition, and they are not used/required, so unsure why.

Reply:
My thinking was to keep the /boot partition generic, but you are right. This is fine